core/src/autogluon/core/__init__.py

```diff
@@ -1,11 +1,29 @@
-from autogluon.common.utils.log_utils import _add_stream_handler 
+from  typing import Any
+from autogluon.common.utils.log_utils import _add_stream_handler
 
 from .dataset import TabularDataset
-from .space import Space, Categorical, Real, Int, Bool
 from . import metrics
 from . import constants
+from .space import Space, Categorical, Real, Int, Bool, spaces
+from .version import __version__
 
 
-_add_stream_handler()
+import warnings
 
-from .version import __version__
+# TODO: Remove deprecation warning for v1.0
+class DeprecatedSpacesWrapper:
+    def __getattr__(self, attr: str) -> Any:
+        import autogluon.common as ag
+
+        if attr in spaces:
+            warnings.warn(
+                "Accessing search spaces as `autogluon.core.space` is deprecated as of v0.8 and won't be supported "
+                "in the next release. Please use `autogluon.common.space` instead."
+            )
```

Comment:
Unsure best practice, but I wonder if we should have a class/function specifically that checks `__version__` and errors if `__version__ >= 0.9`, else warn, as some kind of standardized logic across the repo. Currently we track the deprecations in a bunch of different ways, and can miss removing code / logic that was previously deprecated.

Reply:
This wouldn't be something added in this PR, but something we may want to think about (@yinweisu @tonyhoo @gradientsky )

Reply:
+1. I agree we should think about a general way to handle deprecation

Reply:
We introduced the [`deprecated` decorator](https://github.com/autogluon/autogluon/blob/master/common/src/autogluon/common/utils/deprecated.py) for methods, but I came across a few cases where it is not applicable (e.g., when an argument to a method is renamed).

A simple approach that would cover these edge cases is to always check all the occurrences of the string `deprecate` within the codebase as a part of the release checklist. Though, this is definitely not elegant or scalable.